### ISSUE TITLE
Why you forcing update firebase-tools and NOW our system is not working!!

### ISSUE THREAD

**Issue by rendomnet** [firebase/firebase-tools#5407]
**Our system was perfectly fine and in working condition.**

Firebase forced update us (because of this issue https://github.com/firebase/firebase-tools/issues/5334) to latest version of firebase-tools
and now it is giving us error
```
i  functions: Watching "C:\....\functions\dist" for Cloud Functions...
The system cannot find the path specified.

Error: An unexpected error has occurred.
```
(because of this issue https://github.com/firebase/firebase-tools/issues/4952)

My system was working, and now you are forcing us to upgrade firebase-tools.
and new firebase-tools FORCED us to update firebase-admin and firebase-functions(package.json indicates an outdated version of firebase-functions. Please upgrade using npm install --save firebase-functions@latest in your functions directory.)!
But your new version is NOT WORKING and cant event run.

It's just insane, why we cant use working version of firebase?

```
[debug] [2023-01-09T06:39:03.107Z] Building nodejs source
[debug] [2023-01-09T06:39:03.108Z] Could not find functions.yaml. Must use http discovery
[debug] [2023-01-09T06:39:03.123Z] Error: spawn ./node_modules/.bin/firebase-functions ENOENT
    at notFoundError ***\AppData\Roaming\nvm\v16.19.0\node_modules\firebase-tools\node_modules\cross-spawn\lib\enoent.js:6:26)
    at verifyENOENT ***\AppData\Roaming\nvm\v16.19.0\node_modules\firebase-tools\node_modules\cross-spawn\lib\enoent.js:40:16)
    at ChildProcess.cp.emit ***\AppData\Roaming\nvm\v16.19.0\node_modules\firebase-tools\node_modules\cross-spawn\lib\enoent.js:27:25)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:293:12)
[error] 
[error] Error: An unexpected error has occurred.
```

### [REQUIRED] Environment info

**firebase-tools:**
firebase-tools@10.9.2 pre update
and firebase-tools@latest after update

Mac, Windows

**Platform:**

Mac, Windows

### [REQUIRED] Steps to reproduce

We have a working system on firebase-tools@10.9.2 and "firebase-admin": "10.3.0" "firebase-functions": "3.24.1". And now it is giving error when we deploy

```
i  functions: packaged ***\functions\dist (207.71 KB) for uploading

Error: Failed to fetch Run service undefined
error Command failed with exit code 1.
```

### [REQUIRED] Expected behavior

Successfull deploy

### [REQUIRED] Actual behavior

Forcing us to update firebase-tools and not working after the update

**Comment by JFGHT** [comment]
This was indeed a problem for us too, but keep in mind there are features that have been releasing on Firebase which weren't supported by Firebase-tools (like TTL for example).

Keep an eye on the features you are using.

**Comment by dfidalg0** [comment]
Same issue... I'm using `npm` workspaces. It seems like the `firebase` CLI is trying to use the `firebase-functions` CLI from `functions/node_modules/.bin`, which doesn't exist, since dependencies are installed in the root `node_modules`.

A possible workaround is creating a symlink to `node_modules/.bin` inside `functions/node_modules/.bin`. This solved the problem for me. Gonna try to open a PR to fix this asap.

**Comment by dfidalg0** [comment]
Aparently, this is solved by #5391, but it doesn't seem to made it to npm yet.
